### PR TITLE
:book: hack: generate cli docs

### PR DESCRIPTION
**Pull request by varshaprasad96**
## Related issue(s)

Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,6 +128,9 @@ update-contextual-logging: $(LOGCHECK)
 	UPDATE=true ./hack/verify-contextual-logging.sh
 .PHONY: update-contextual-logging
 
+generate:
```
Comment (sttts) [inline Makefile:131]: nit: docs ?

File: hack/generate/cli-doc/gen-cli-doc.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2021 The KCP Authors.
```
Comment (sttts) [inline hack/generate/cli-doc/gen-cli-doc.go:2]: 2022

**Comment by stevekuznetsov** [comment]
@varshaprasad96 please use `make update-contextual-logging` to pass lint.

**Comment by varshaprasad96** [comment]
Since we are fine with having doc generation for cli, I'll modify the PR to include all cobra commands, and have this run in CI. 
/hold

**Comment by varshaprasad96** [comment]
@avinal Thoughts on merging this PR here. Can they later be modified to be Hugo compatible? I'm not sure of what changes are required to do so.

**Comment by avinal** [comment]
> @avinal Thoughts on merging this PR here. Can they later be modified to be Hugo compatible? I'm not sure of what changes are required to do so.

Hi Varsha, I am still looking into the options to integrate everything with Hugo, should take a day or two. I think we can merge this for now. In any case we can always open a PR.

**Review by stevekuznetsov — APPROVED** [review]

**Review by avinal — APPROVED** [review]
/lgtm

Tested, works as expected.

**Pull request merged**